### ISSUE TITLE
Implement HSTS support for communication with servers

### ISSUE THREAD

**Issue by PatoBeltran** [NuGet/Home#6940]
Opening this issue after a suggestion from @blowdart to add support for HSTS in our clients to improve security when communicating with servers.

**Comment by maartenba** [comment]
Have discussed this with @blowdart in the past and he thought this was not something the NuGet client should be doing? Anyway, I love the idea!

**Comment by blowdart** [comment]
What, I'm not allowed to change my mind? To set expectations here it'd just be the header, there would be no consideration of the preload list.

**Comment by maartenba** [comment]
Makes sense. And I am really happy you changed your mind, thanks for that :-)
